impl/src/main/java/org/ehcache/internal/store/heap/OnHeapStore.java

```diff
@@ -366,8 +366,17 @@ public void clear() throws CacheAccessException {
     map.clear();
   }
 
-  @Override
-  public void invalidate() throws CacheAccessException {
+  private void invalidate() throws CacheAccessException {
+    if(map.map != null) {
+      for (K k : map.map.keySet()) {
+        invalidate(k);
+      }
+    }
+    if(map.keyCopyMap != null) {
+      for(OnHeapKey<K> key : map.keyCopyMap.keySet()) {
+        invalidate(key.getActualKeyObject());
+      }
+    }
```

Comment:
This can't possibly happen! See the discussion on @lorban PR, this method _should_ be a plain clear!